### PR TITLE
Merge Edge Feature Branch into Master

### PR DESCRIPTION
**Pull request by elvinm**
# Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist:

- [x] All new and existing tests passed.
- [x] The code follows the code style and conventions of this project.
- [x] The change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

# Description of the change
Merging all prior edge branches & commits into master.

### CODE REVIEW

File: config/test/com/microsoft/azure/iotsolutions/uiconfig/services/StorageTest.java [hunk 1]
```diff
@@ -350,11 +352,93 @@ public void updateDeviceGroupAsyncTest() throws BaseException, ExecutionExceptio
...
+        List<Package> results = Lists.newArrayList(storage.getAllPackagesAsync().toCompletableFuture().get());
+        assertEquals(5, results.size());
+
+        for (int i = 0; i < results.size(); i++) {
+            Package result = results.get(i);
+            ValueApiModel value = items.stream().filter(m -> m.getKey().equals(result.getId())).findFirst().get();
+            Package pkg = Json.fromJson(Json.parse(value.getData()), Package.class);
+            assertEquals(pkgName + i, pkg.getName());
+            assertEquals(content + i, pkg.getContent());
+
```
Comment (ppathan) [inline config/test/com/microsoft/azure/iotsolutions/uiconfig/services/StorageTest.java]: nit: extra line.

File: iothub-manager/app/com/microsoft/azure/iotsolutions/iothubmanager/webservice/v1/models/TwinPropertiesApiModel.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
+package com.microsoft.azure.iotsolutions.iothubmanager.webservice.v1.models;
```
Comment (ppathan) [inline iothub-manager/app/com/microsoft/azure/iotsolutions/iothubmanager/webservice/v1/models/TwinPropertiesApiModel.java]: // Copyright (c) Microsoft. All rights reserved.

File: iothub-manager/test/com/microsoft/azure/iotsolutions/iothubmanager/services/DevicesUnitTest.java [hunk 3]
```diff
@@ -0,0 +1,156 @@
+package com.microsoft.azure.iotsolutions.iothubmanager.services;
```
Comment (ppathan) [inline iothub-manager/test/com/microsoft/azure/iotsolutions/iothubmanager/services/DevicesUnitTest.java]: // Copyright (c) Microsoft. All rights reserved.

File: iothub-manager/test/com/microsoft/azure/iotsolutions/iothubmanager/services/DevicesUnitTest.java [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.ArgumentMatchers.*;
+import static org.mockito.Mockito.when;
+
+@RunWith(JUnitParamsRunner.class)
+public class DevicesUnitTest {
```
Comment (ppathan) [inline iothub-manager/test/com/microsoft/azure/iotsolutions/iothubmanager/services/DevicesUnitTest.java]: // Arrange ...

File: config/test/com/microsoft/azure/iotsolutions/uiconfig/services/StorageTest.java [hunk 5]
```diff
@@ -350,11 +352,93 @@ public void updateDeviceGroupAsyncTest() throws BaseException, ExecutionExceptio
...
         DeviceGroup result = storage.updateDeviceGroupAsync(groupId, group, etagOld).toCompletableFuture().get();
         assertEquals(result.getId(), groupId);
         assertEquals(result.getDisplayName(), displayName);
         assertEquals(result.getConditions(), conditions);
         assertEquals(result.getETag(), etagNew);
     }
+
+    @Test
+    @Category({UnitTest.class})
+    public void addPackageTest() throws BaseException, ExecutionException, InterruptedException {
```
Comment (ppathan) [inline config/test/com/microsoft/azure/iotsolutions/uiconfig/services/StorageTest.java]: // Arrange ... in this file for new tests.

File: iothub-manager/app/com/microsoft/azure/iotsolutions/iothubmanager/webservice/v1/models/TwinPropertiesApiModel.java (hunk 2 above)
Comment (elvinm) [reply iothub-manager/app/com/microsoft/azure/iotsolutions/iothubmanager/webservice/v1/models/TwinPropertiesApiModel.java]: Added

File: config/test/com/microsoft/azure/iotsolutions/uiconfig/services/StorageTest.java (hunk 1 above)
Comment (elvinm) [reply config/test/com/microsoft/azure/iotsolutions/uiconfig/services/StorageTest.java]: Fixed

File: iothub-manager/test/com/microsoft/azure/iotsolutions/iothubmanager/services/DevicesUnitTest.java (hunk 3 above)
Comment (elvinm) [reply iothub-manager/test/com/microsoft/azure/iotsolutions/iothubmanager/services/DevicesUnitTest.java]: Added

File: iothub-manager/test/com/microsoft/azure/iotsolutions/iothubmanager/services/DevicesUnitTest.java (hunk 4 above)
Comment (elvinm) [reply iothub-manager/test/com/microsoft/azure/iotsolutions/iothubmanager/services/DevicesUnitTest.java]: Done

File: config/test/com/microsoft/azure/iotsolutions/uiconfig/services/StorageTest.java (hunk 5 above)
Comment (elvinm) [reply config/test/com/microsoft/azure/iotsolutions/uiconfig/services/StorageTest.java]: Done

**Pull request merged**